### ISSUE TITLE
Suggestion: Introduce JS Linter

### ISSUE THREAD

**Issue by vfonic** [solidusio/solidus#1719]
The solidus js/coffeescript code is growing bigger and bigger every day. There doesn't seem to be a unified code style or something that would enforce it.

Things like:
```js
jQuery(function($) {
$(document).ready(function(){
(function() {
```

or

```js
Spree.formatMoney = function(
window.Spree.advanceOrder = function() {
```

...using semicolons or not, spacing between `function()` and `{` etc.

If we follow the same style, and I'd add, use coffeescript or javascript exclusively, not intertwined, the code would be easier to read, and also we'd avoid some potential issues like global function overriding, automatic semicolon insertion bugs, etc.

I propose to use [ESLint](http://eslint.org/). It's the most popular js linter at the moment, very configurable and extendable and has predefined configurations, such as [JS standard style](https://www.npmjs.com/package/eslint-config-standard) or [airbnb style](https://www.npmjs.com/package/eslint-config-airbnb) (which is very popular).

**Comment by tvdeyen** [comment]
Generally this is a great idea and as we already use HoundCI for Ruby linting we could enable their JS linting as well. 

BUT: As with every code style guidance there are religious wars to expect. That's why we want to reduce the linting to the absolute minimum. We don't want to run single vs double quotes, leading comma vs trailing comma and so on and so forth debates. 

Also, we need to update the current code before we enable Hound JS linting as we would have lots of red PRs then. 

We happely accept PRs that help us updating the exisiting code. After that is done there is nothing stopping us from enabling a automatic linter. 

Thanks for mentioning

**Comment by vfonic** [comment]
We should also agree about which rules the linter should use then. I just created a PR for CoffeeScript, but not sure that the configuration is what you guys are looking-for/agree-on?

**Comment by vfonic** [comment]
I personally think we should convert those coffeescript files to javascript and use js exclusively.

**Comment by vfonic** [comment]
Just ran `rubocop` and here's the output:
```
1392 files inspected, 1154 offenses detected
```

**Comment by jhawthorn** [comment]
Fixed by #1736
